lib/fog/cloudsigma/models/snapshot.rb

```diff
@@ -0,0 +1,76 @@
+require 'fog/cloudsigma/nested_model'
+require 'fog/cloudsigma/models/volume'
+
+module Fog
+  module Compute
+    class CloudSigma
+      class Snapshot < Fog::CloudSigma::CloudsigmaModel
+        identity :uuid
+
+        attribute :timestamp, :type => :string
+        attribute :tags
+        attribute :status, :type => :string
+        attribute :resource_uri, :type => :string
+        attribute :permissions, :type => :array
+        attribute :owner
+        attribute :name, :type => :string
+        attribute :meta
+        attribute :grantees, :type => :array
+        attribute :drive
+        attribute :allocated_size, :type => :integer
+        
+        def save
+          if persisted?
+            update
+          else
+            create
+          end
+        end
+
+        def create
+          requires :name, :drive
+          data = attributes
+
+          response = service.create_snapshot(data)
+          new_attributes = response.body['objects'].first
+          merge_attributes(new_attributes)
+        end
+
+        def update
+          requires :identity, :name
+
+          data = attributes()
+
+          response = service.update_snapshot(identity, data)
+          new_attributes = response.body
+          merge_attributes(new_attributes)
+        end
+
+        def destroy
+          requires :identity
+
+          service.delete_snapshot(identity)
+
+          true
```

Comment:
But I am afraid that way user will not be able to know what caused this 'false' reply. It could be a simple 404 (meaning he was trying to delete a snapshot that did not exist) or it could be 5xx error message signalling of API server failure. These exceptions are used in this library just for that: they convey the reason of failure with type and meaningful error message as with the example I posted in the previous message.